distributed/dashboard/scheduler.py

```diff
@@ -112,41 +112,45 @@
 }
 
 
-template_variables: dict = {
-    "pages": [
-        "status",
-        "workers",
-        "tasks",
-        "system",
-        "profile",
-        "graph",
-        "groups",
-        "info",
-    ],
-    "plots": [
-        {
-            "url": x.strip("/"),
-            "name": " ".join(x.strip("/").split("-")[1:])
-            .title()
-            .replace("Cpu", "CPU")
-            .replace("Gpu", "GPU"),
-        }
-        for x in applications
-        if "individual" in x
-    ]
-    + [{"url": "hardware", "name": "Hardware"}],
-}
-template_variables["plots"] = sorted(
-    template_variables["plots"], key=lambda d: d["name"]
-)
+@memoize
+def template_variables():
+    from distributed.diagnostics.nvml import device_get_count
 
-if NVML_ENABLED:
-    template_variables["pages"].insert(4, "gpu")
+    template_variables: dict = {
+        "pages": [
+            "status",
+            "workers",
+            "tasks",
+            "system",
+            "profile",
+            "graph",
+            "groups",
+            "info",
+        ],
+        "plots": [
+            {
+                "url": x.strip("/"),
+                "name": " ".join(x.strip("/").split("-")[1:])
+                .title()
+                .replace("Cpu", "CPU")
+                .replace("Gpu", "GPU"),
+            }
+            for x in applications
+            if "individual" in x
+        ]
+        + [{"url": "hardware", "name": "Hardware"}],
+    }
+    template_variables["plots"] = sorted(
+        template_variables["plots"], key=lambda d: d["name"]
+    )
+    if device_get_count() > 0:
```

Comment:
I know this was here before but can you add a comment about why we are inserting into the fifth position ?